src/query/v2/interpreter.cpp

```diff
@@ -940,6 +942,7 @@ PreparedQuery PrepareExplainQuery(ParsedQuery parsed_query, std::map<std::string
                                   InterpreterContext *interpreter_context,
                                   msgs::ShardRequestManagerInterface *shard_request_manager,
                                   utils::MemoryResource *execution_memory) {
+  shard_request_manager->StartTransaction();
```

Comment:
I had to solve the same problem for the profile query. I think the proper way to fix this is done in https://github.com/memgraph/memgraph/pull/617/commits/5939fb2b0cadd14bfee3081f57625ab709e533ea

Reply:
I slightly altered this. I move the `StartTransaction` at the top of the `Prepare` function. That way, the plan rewritter has access to `NameId` mappers.

Reply:
Why though? The planning happens in the `PrepareCypherQuery` function and even with my changes the planner/rewriter has proper access to the shard map. I think the difference between the current version and my changes (that works better in my opinion, because it only starts a transaction when it is necessary, not in case of `BEGIN`/`COMMIT`/`ROLLBACK` queries) is small, but has advantages.

Reply:
I was wrong, I moved it back